alanEappen/assignments/slidingwindow/lc209/lc209.js

```diff
@@ -1,10 +1,33 @@
+
+// arr is a postive integer array 
+// s is a given number to compare 
+// to find the smallest subarray that is equal or greater than s
 function smallest_subarray_with_given_sum(s, arr) {
-  // TODO: Write code here
+  let totalSum = 0 //calculate the sum
+  minLength = arr.length //making the minLength to the max size of the array
+  windowStart = 0
+
+  if (arr.length == 0) {
+    return 0;
+  }
+
+  for (let i = 0; i < arr.length; i++) {
+    totalSum += arr[i]; // here we add the next element
+    // here we are shrinking window as small possible until the sum is smaller than s
+    while (totalSum >= s) {
+      minLength = Math.min(minLength, i - windowStart + 1);
+      totalSum -= arr[windowStart];
+      windowStart++;
+    }
+  }
+
+  return minLength;
```

Comment:
Great use of comments & code is readable, great stuff! Remember you gotta be nice and type it twice! Where's the repetition @ ??